### PR TITLE
Add MatrixTrait::try_invert()

### PR DESCRIPTION
**Pull request by federicomenaquintero**
As discussed in https://github.com/gtk-rs/cairo/issues/116

I also added PartialEq to Matrix, so I can compare it in tests :)

### CODE REVIEW

File: src/matrices.rs [hunk 1]
```diff
@@ -84,6 +107,19 @@ impl MatrixTrait for Matrix {
...
+    fn try_invert(&self) -> Result<Matrix, Status> {
+        let mut matrix = *self;
+
+        let result = unsafe {
+            ffi::cairo_matrix_invert(&mut matrix)
+        };
+
+        match result {
+            Status::Success => Ok(matrix),
+            _ => Err(result)
```
Comment (GuillaumeGomez) [inline src/matrices.rs]: Nit: missing a comma at the end of this line.

**Comment by GuillaumeGomez** [comment]
Excellent, thanks a lot! I'm strongly in favor of this PR.

Is it ok for you as well @EPashkin?

File: src/matrices.rs (hunk 1 above)
Comment (federicomenaquintero) [inline src/matrices.rs]: Just pushed another commit to add the comma.  I can squash it and force-push it if you like... do we use this same policy as for rust-lang pull requests?

Comment (GuillaumeGomez) [inline src/matrices.rs]: Just for small nits like this, so just squash last commit please. For the others it's fine.

**Comment by federicomenaquintero** [comment]
Squashed the missing comma, and added docs for invert() as well.

**Comment by GuillaumeGomez** [comment]
Thanks a lot! Just waiting for @EPashkin's opinion now.

**Comment by EPashkin** [comment]
@GuillaumeGomez Code looks good.
But Travis failed due presence of docs.
It need moved to https://github.com/gtk-rs/lgpl-docs ?

**Comment by GuillaumeGomez** [comment]
Well, that's the strange part. I think we never had the case.

@federicomenaquintero: Can you move it in the lgpl-docs repo as @EPashkin suggested please?

**Comment by federicomenaquintero** [comment]
Uh, how do I build the docs from lgpl-docs? :)  I'd like to see if I'm writing the right thing there.

**Comment by EPashkin** [comment]
To build docs you need add it from https://github.com/gtk-rs/lgpl-docs with
```
cargo build --features embed-lgpl-docs
```
then build generate docs as always:
```
cargo doc
```

**Comment by federicomenaquintero** [comment]
I've removed the docs from my cairo branch, and added them to lgpl-docs in https://github.com/gtk-rs/lgpl-docs/pull/27

**Comment by EPashkin** [comment]
👍 
@federicomenaquintero Thanks.

**Comment by GuillaumeGomez** [comment]
Thanks!

**Pull request merged**